### PR TITLE
*: upgrade go1.21.3

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #47441

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by hawkingrei** [comment]
Please @bb7133 decide when to merge.

**Comment by wuhuizuo** [comment]
Maybe we should upgrade golang to `1.21.3`, it will be released at Oct 10.
Ref:  https://groups.google.com/g/golang-announce/c/XIujkcMT56U

File: WORKSPACE [hunk 1]
```diff
@@ -64,7 +64,7 @@ go_download_sdk(
         "https://mirrors.aliyun.com/golang/{}",
         "https://dl.google.com/go/{}",
     ],
-    version = "1.21.1",
+    version = "1.21.2",
```
Comment (wuhuizuo) [inline WORKSPACE:67]:
```suggestion
    version = "1.21.3",
```

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**